examples/state.rs

```diff
@@ -5,20 +5,18 @@
 #![no_main]
 #![no_std]
 
-#[macro_use(entry, exception)]
 extern crate cortex_m_rt as rt;
 extern crate panic_semihosting;
 
-// the program entry point
-entry!(main);
+use rt::{entry, exception};
 
+#[entry]
 fn main() -> ! {
     loop {}
 }
 
 // exception handler with state
-exception!(SysTick, sys_tick, state: u32 = 0);
-
-fn sys_tick(state: &mut u32) {
-    *state += 1;
+#[exception(SysTick, static STATE: u32 = 0)]
```

Comment:
An alternative syntax would be to have the second argument be a string. i.e.

``` rust
#[exception(SysTick, "static STATE: u32 = 0")]
fn sys_tick() {}
```

But the current syntax seems to parse without problems and have better spans for error messages. e.g.

``` rust
#[exception(SysTick, static STATE: u32 = true)]
//                                       ~~~~ error: expected `u32`, found `bool`
fn sys_tick() {}
```